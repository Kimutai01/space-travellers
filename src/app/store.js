import { configureStore } from '@reduxjs/toolkit';
import missionsReducer from '../redux/missions/missions';
import rocketReducer from '../redux/rocket/rocket';

const store = configureStore({
  reducer: {
    missions: missionsReducer,
    rockets: rocketReducer,
  },
});

export default store;
