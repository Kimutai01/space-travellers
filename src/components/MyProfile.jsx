import { useSelector } from 'react-redux';

const MyProfile = () => {
  const missionsL = useSelector((state) => state.missions);
  const newMissions = missionsL.filter((mission) => mission.joined);
  const rocketsL = useSelector((state) => state.rockets);
  const newRockets = rocketsL.filter((rocket) => rocket.canceled);
  return (
    <div>
      <div>
        <h2>My Missions</h2>
        <ul>
          {newMissions.length >= 1 ? (
            newMissions.map((mission) => (
              <li key={mission.name}>
                {mission.name} by {mission.manufacturer}
              </li>
            ))
          ) : (
            <li>You are not part of any mission</li>
          )}
        </ul>
      </div>
      <div>
        <h2>My Rockets</h2>
        <ul>
          {newRockets.length >= 1 ? (
            newRockets.map((rocket) => (
              <li key={rocket.rocket_name}>{rocket.rocket_name}</li>
            ))
          ) : (
            <li>No Rockets available </li>
          )}
        </ul>
      </div>
    </div>
  );
};

export default MyProfile;
