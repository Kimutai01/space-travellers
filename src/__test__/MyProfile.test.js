import MyProfile from '../components/MyProfile';
import { Provider } from 'react-redux';
import store from '../app/store';
import TestRenderer from 'react-test-renderer';
test('My profile component testing', () => {
  const TREE = TestRenderer.create(
    <Provider store={store}>
      <MyProfile />
    </Provider>
  );
  expect(TREE).toMatchSnapshot();
});
