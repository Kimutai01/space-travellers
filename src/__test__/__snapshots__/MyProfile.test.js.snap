// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My profile component testing 1`] = `
<div
  className="profile-section"
>
  <div
    className="mission-div"
  >
    <h2>
      My Missions
    </h2>
    <ul>
      <li>
        You are not part of any mission
      </li>
    </ul>
  </div>
  <div
    className="mission-div"
  >
    <h2>
      My Rockets
    </h2>
    <ul>
      <li>
        No Rockets available 
      </li>
    </ul>
  </div>
</div>
`;
